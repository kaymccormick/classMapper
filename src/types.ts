import { namedTypes } from 'ast-types/gen/namedTypes';
import EntityCore from "classModel/lib/src/entityCore";
export type ModuleSpecifier = string;
import { Connection } from 'typeorm';
import { Logger } from 'winston';
import {RestClient} from './RestClient';

export interface PromiseResult<T> {
  id: string;
  success: boolean;
  hasResult: boolean;
  result?: T;
  error?: Error;
}

export interface Args<T> {
    connection: Connection;
    restClient: RestClient;
    logger: Logger;
    caller?: T;
}

export interface HandleAst {
    (args: Args<any>, project: EntityCore.Project,fname: string,ast: namedTypes.File): Promise<void>;
}

export interface ImportContext {
    module: ModuleSpecifier;
    moduleEntity: EntityCore.Module;
}

export interface HandleImportSpecifier {
    (importContext: ImportContext, importModuleName: string, localName: string, exportedName?: string, isDefault?: boolean, isNamespace?: boolean): Promise<void>;
}
