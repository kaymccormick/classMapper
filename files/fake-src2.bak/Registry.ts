import {Entity as OrmEntity} from "typeorm";
import {Column, PrimaryColumn} from "typeorm";

/* empty */
