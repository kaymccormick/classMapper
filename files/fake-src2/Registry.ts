/** @uuid 2407aae2-06ec-4afd-b828-d5f22e7c6433
*/
import { Entity as OrmEntity } from "typeorm";

import { Column, PrimaryColumn } from "typeorm";/* empty */