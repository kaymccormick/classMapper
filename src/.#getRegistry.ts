jade@pluto.heptet.us.13920:1559367296