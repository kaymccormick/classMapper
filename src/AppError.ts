export default class AppError extends Error {
    public constructor(message: string, public readonly id: string=message) {
        super(message);
    }
}
