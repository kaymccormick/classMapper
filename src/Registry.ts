import * as fs from 'fs';
import { CreateModuleFunction, createModule} from './Factory';
import {Module} from "./Module";
import { Map } from 'immutable';
import {ModulePojo} from "./types";
export interface Registry {
    init(): void;
    registerModule(module: Module): void;
    registerClass(classKind: string): void;
    getModule(name: string, create?: boolean): Module;
}

interface SimpleRegistryArgs {
    runId?: number;
    load?: boolean;
}

interface SimpleRegistryPojo {
    runId?: number;
    modules: Map<string, ModulePojo>;
}

interface SimpleRegistryData {
    runId?: number;
    modules: Map<string, Module>;
}

class SimpleRegistryDataImpl implements SimpleRegistryData {
    public modules: Map<string, Module> = Map<string, Module>();
    public runId?: number;

    public constructor(runId: number) {
        this.runId = runId;
    }
}

export class SimpleRegistry implements Registry {
    private runId?: number;
    private data: SimpleRegistryData = new SimpleRegistryDataImpl(0);
    private load: boolean = false;

    // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
    public init(): void {
        const data: SimpleRegistryPojo = JSON.parse(fs.readFileSync('registry.json', {encoding: 'utf-8'}));
        const modules2 = Map<string, ModulePojo>(data.modules);
        this.data = { runId: data.runId,
        modules: modules2.map((v: ModulePojo): Module => Module.fromPojo(v)),
        };

        if (!this.load && (data.runId === undefined || data.runId !== this.runId)) {
            console.log('resetting data');
            this.initBareData();
        }

    }

    public registerClass(classKind: string): void {
        throw new Error("Method not implemented.");
    }

    // private modules: Map<string, Module> = Map<string, Module>();

    public constructor(args: SimpleRegistryArgs) {
        this.runId = args.runId;
        this.load = args.load || false;

    }

    public registerModule(module: Module) {
    }

    public getModule(name: string, create: boolean = false): Module {
        if(this.data.modules.has(name)) {
            const newVar = this.data.modules.get(name);
            if(!newVar) {
                throw new Error('undefined module');
            }
            return newVar
        } else if(create) {
            const module = createModule(name);
            this.data.modules = this.data.modules.set(name, module);
            return module;
        }else {
            throw new Error(`nonexistent module ${name}`)
        }
    }

    private initBareData() {
        this.data = {modules: Map<string, Module>(), runId: this.runId};
    }

    public save() {
        const modules = this.data.modules.map<ModulePojo>((v) => {
            if(!v.toPojo) {
                throw new Error('no pojo');
            }
            return v.toPojo();
        }).toJS();
        this.data.modules.forEach((v, k) => {
            //console.log(v);
            //console.log(v.toPojo())
        });
        //const modules = {}
        //const modules = this.data.modules.map((v) => v.toPojo()).toJS();
        const data = { runId: this.data.runId,
            modules };
        fs.writeFileSync('registry.json', JSON.stringify(data, null, 4), 'utf-8');
    }
}
