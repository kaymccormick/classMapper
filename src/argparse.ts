import {ArgumentParser} from 'argparse';

export function getDefaultArgumentParser(): ArgumentParser {
const parser = new ArgumentParser({});
parser.addArgument(['--projectId'], { help: 'project ID to operate on' });
parser.addArgument(['--consoleLogLevel'], { help: 'console log level' });
parser.addArgument(['--config'], { help: 'specify config file'});
return parser;
}
