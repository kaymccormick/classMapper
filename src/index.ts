export { createConnection } from './TypeOrm/Factory';
import EntityCore from 'classModel/lib/src/entityCore';
export { EntityCore };
