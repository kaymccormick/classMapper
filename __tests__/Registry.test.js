import { SimpleRegistry } from '../src/Registry';
import {Registry} from "../src/types";

test('1', () => {
    const r = new SimpleRegistry({runId: 0});
    r.init();
    const module = r.getModule('test', true);
    expect(r.getModule('test')).toStrictEqual(module);
    r.save();
});
