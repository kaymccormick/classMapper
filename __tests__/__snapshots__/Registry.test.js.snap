// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2 1`] = `
SimpleRegistry {
  "data": Object {
    "modules": Immutable.Map {
      "test": Module {
        "classes": Immutable.Map {},
        "defaultExport": undefined,
        "exports": Immutable.Map {},
        "imports": Immutable.Map {},
        "interfaces": Immutable.Map {},
        "name": "test",
        "references": Immutable.Map {},
      },
    },
    "runId": 0,
  },
  "dataImpl": SimpleRegistryDataImpl {
    "_modules": Immutable.Map {},
    "runId": 0,
  },
  "load": true,
  "runId": undefined,
}
`;
